test/e2e/alertmanager_test.go

```diff
@@ -135,6 +135,130 @@ enableUserWorkload: true`,
 	}
 }
 
+func TestAlertmanagerPolicyHeaders(t *testing.T) {
+	const (
+		config             = "alertmanagerMain:"
+		userWorkloadConfig = ""
+		amName             = "main"
+	)
+	amNamespace := f.Ns
+
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      framework.ClusterMonitorConfigMapName,
+			Namespace: f.Ns,
+			Labels: map[string]string{
+				framework.E2eTestLabelName: framework.E2eTestLabelValue,
+			},
+		},
+		Data: map[string]string{
+			"config.yaml": config,
+		},
+	}
+	f.MustCreateOrUpdateConfigMap(t, cm)
+	t.Cleanup(func() {
+		f.MustDeleteConfigMap(t, cm)
+	})
+
+	testAlertmanagerReady(t, amName, amNamespace)
+
+	// The tenancy port (9092) is only exposed in-cluster so we need to use
+	// port forwarding to access kube-rbac-proxy.
+	host, cleanUp, err := f.ForwardPort(t, amNamespace, fmt.Sprintf("alertmanager-%s", amName), 9092)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(cleanUp)
+
+	testAlertmanagerPolicyHeaders(t, host)
+}
+
+func testAlertmanagerPolicyHeaders(t *testing.T, host string) {
+	t.Helper()
+
+	ctx := context.Background()
+	const testNs = "policy-headers-e2e-test"
+
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testNs,
+			Labels: map[string]string{
+				framework.E2eTestLabelName: framework.E2eTestLabelValue,
+			},
+		},
+	}
+	ns, err := f.KubeClient.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		foreground := metav1.DeletePropagationForeground
+		if err := f.KubeClient.CoreV1().Namespaces().Delete(ctx, testNs, metav1.DeleteOptions{PropagationPolicy: &foreground}); err != nil {
+			t.Logf("err deleting namespace %s: %v", testNs, err)
+		}
+	})
+
+	sa := "editor"
+	cr := "monitoring-rules-edit"
+
+	_, err = f.CreateServiceAccount(testNs, sa)
```

Comment:
instead of creating a new service account + bindings, we could reuse the [e2e service account](https://github.com/openshift/cluster-monitoring-operator/blob/d4fa84c8be2e3e20f6d65829022d793339b0e491/test/e2e/framework/framework.go#L59) which should already have all RBACs.

Reply:
thanks, working on it

Reply:
If I change token to:

`		token, err := f.GetServiceAccountToken(namespaceName, e2eServiceAccount)
		if err != nil {
			return err
		}`

I am getting: 
`mariofar respo &{403 Forbidden 403 HTTP/1.1 1 1 map[Content-Length:[142] Content-Type:[text/plain; charset=utf-8] Date:[Mon, 03 Jul 2023 16:56:11 GMT] X-Content-Type-Options:[nosniff]] 0xc000525d40 142 [] false false map[] 0xc0000da900 0xc0001454a0}
`

Reply:
Fixed.